drivers/timer/cavs_timer.c

```diff
@@ -195,9 +196,6 @@ void smp_timer_init(void)
 	 * FIXME: Done in this way because we don't have an API
 	 * to enable interrupts per CPU.
 	 */
-	sys_set_bit(DT_REG_ADDR(DT_NODELABEL(cavs0))
-			+ CAVS_ICTL_INT_CPU_OFFSET(arch_curr_cpu()->id)
-			+ 0x04,
-		    22 + TIMER);
+	CAVS_INTCTRL[arch_curr_cpu()->id].l2.clear = CAVS_L2_DWCT0;
```

Comment:
It's still correct.  These two bits of code are identical (the new API just names the register being used instead of using an offset).  The complaint here is that the cavs_ictl interrupt controller driver only understands global masking and doesn't hook the per-core mask registers.  So the individual device drivers need to do it manually.  Definitely a spot for cleanup in the future.